### DISCUSSION TITLE
Noob Question: ESP32 WROOM32 get enery data without WLAN and save total energy

### DISCUSSION THREAD

**Discussion by zutewas** [klaasnicolaas/home-assistant-glow#475] [Q&A]
Hello, thanks for this project. I'm new in this topic and try to orientate me.

I bought a ESP32 WROOM32 USB-C ESP32-DevKitC 32Mb/4MB 30 38 Pin Development Board Adapter and a 3/4 Pin Diode Sensor Modul LM393-N. It works with Home Assistant and I get data in my entities.

It is possible to use this setup without WLAN? At night my WLAN is diabled and I lost the energy data. The ESP32 is connected with the Raspberry 3 where HA is running. Can I get the enery meter data via USB too?

After lost WLAN connection for hours I can't see my data on web interface of ESP32 anymore. My calibrated values "Reset Value - Total Energy kWh" are reseted.

Can anybody give me a approach what I can do to save my data persistent and using the ESP32 directly via USB on my Raspberry where HA is running too. Thank for any hints :-)

**Comment by klaasnicolaas** [comment]
> After lost WLAN connection for hours I can't see my data on web interface of ESP32 anymore.

If ESPHome loses connection to your WiFi network, it will set up its own WiFi AP, this will also reboot the glow and reset the values. I think it is better to ask this question via the ESPHome channels, as your question fits better there.

./Klaas
